src/v/cloud_topics/reconciler/reconciler.cc

```diff
@@ -112,10 +112,11 @@ ss::future<> reconciler::reconciliation_loop() {
      */
     constexpr std::chrono::seconds poll_frequency(10);
 
+    ss::lowres_clock::duration next_wait = poll_frequency;
     while (!_gate.is_closed()) {
         try {
             co_await _control_sem.wait(
-              poll_frequency, std::max(_control_sem.current(), size_t(1)));
+              next_wait, std::max(_control_sem.current(), size_t(1)));
```

Comment:
Seems good. An alternative policy is to only wait `poll_frequency` if there was no work todo. But maybe that's non ideal for low throughput systems as it will result in many small L1 objects.